x/dex/spec/README.md

```diff
@@ -1,21 +1,46 @@
 ## Abstract
-
-TODO: Populate dex README Contents below.
-
-## Contents
+`dex` module is responsible for matching orders for registered contracts.
 
 ## Concepts
-
-## State
-
-## Messages
-
-## Events
+### Frequent Batch Auction
+The traditional implementation of exchange logic would look something like the following:
+1. User A sends an order placement transaction
+2. User B sends another order placement transaction
+3. User A's transaction is processed by matching it against the order book state and settle accordingly
+4. User B's transaction is processed similarly
+
+Sei's `dex` module takes a different approach:
```

Comment:
can we specify the difference between limit order processing and market order processing here? probably elaborates the step 1-5 here would be good: https://github.com/sei-protocol/sei-chain/blob/master/x/dex/contract/execution.go#L59-L74